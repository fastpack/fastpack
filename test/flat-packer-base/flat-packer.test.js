const { loadFpackBundle } = require('../helpers.js');

process.chdir(__dirname);

test('flat packer with single js', async () => {
  expect(
    await loadFpackBundle({ input: 'index-single.js', target: 'app' })
  ).toMatchSnapshot('target: app');
  expect(
    await loadFpackBundle({ input: 'index-single.js', target: 'esm' })
  ).toMatchSnapshot('target: esm');
  expect(
    await loadFpackBundle({ input: 'index-single.js', target: 'cjs' })
  ).toMatchSnapshot('target: cjs');
});

test('flat packer with only esm', async () => {
  expect(
    await loadFpackBundle({ input: 'index-esm.js', target: 'app' })
  ).toMatchSnapshot('target: app');
  expect(
    await loadFpackBundle({ input: 'index-esm.js', target: 'esm' })
  ).toMatchSnapshot('target: esm');
  expect(
    await loadFpackBundle({ input: 'index-esm.js', target: 'cjs' })
  ).toMatchSnapshot('target: cjs');
});

test('flat packer with only cjs', async () => {
  expect(
    await loadFpackBundle({ input: 'index-cjs.js', target: 'app' })
  ).toMatchSnapshot('target: app');
  expect(
    await loadFpackBundle({ input: 'index-cjs.js', target: 'esm' })
  ).toMatchSnapshot('target: esm');
  expect(
    await loadFpackBundle({ input: 'index-cjs.js', target: 'cjs' })
  ).toMatchSnapshot('target: cjs');
});

test('flat packer with esm in cjs', async () => {
  expect(
    await loadFpackBundle({ input: 'index-esm-in-cjs.js', target: 'app' })
  ).toMatchSnapshot('target: app');
  expect(
    await loadFpackBundle({ input: 'index-esm-in-cjs.js', target: 'esm' })
  ).toMatchSnapshot('target: esm');
  expect(
    await loadFpackBundle({ input: 'index-esm-in-cjs.js', target: 'cjs' })
  ).toMatchSnapshot('target: cjs');
});

test('flat packer with cjs in esm', async () => {
  expect(
    await loadFpackBundle({ input: 'index-cjs-in-esm.js', target: 'app' })
  ).toMatchSnapshot('target: app');
  expect(
    await loadFpackBundle({ input: 'index-cjs-in-esm.js', target: 'esm' })
  ).toMatchSnapshot('target: esm');
  expect(
    await loadFpackBundle({ input: 'index-cjs-in-esm.js', target: 'cjs' })
  ).toMatchSnapshot('target: cjs');
});
