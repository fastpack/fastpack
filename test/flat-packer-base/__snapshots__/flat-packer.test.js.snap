// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flat packer with cjs in esm: target: app 1`] = `
"(function() {
var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$cjs$$_$$exports */

let $n__dep$$_$$cjs$$_$$exports = { exports: {}};$n__dep$$_$$cjs$$_$$exports.exports.a = 1;
$n__dep$$_$$cjs$$_$$exports.exports.b = 2;

/* dep$$_$$cjs$$_$$module */

let $n__dep$$_$$cjs$$_$$module = { exports: {}};$n__dep$$_$$cjs$$_$$module.exports = {
  a: 1,
  b: 2
};

/* index$$_$$cjs$$_$$in$$_$$esm */





if ($n__dep$$_$$cjs$$_$$exports.exports.a < $n__dep$$_$$cjs$$_$$exports.exports.b) {
  console.log($n__dep$$_$$cjs$$_$$exports.exports.a, $n__dep$$_$$cjs$$_$$exports.exports.b);
}
if ($n__dep$$_$$cjs$$_$$module.exports.a < $n__dep$$_$$cjs$$_$$module.exports.b) {
  console.log($n__dep$$_$$cjs$$_$$module.exports);
}
if ($n__dep$$_$$cjs$$_$$exports.exports.a < $n__dep$$_$$cjs$$_$$exports.exports.b) {
  console.log($n__dep$$_$$cjs$$_$$exports.exports);
}

const $e__index$$_$$cjs$$_$$in$$_$$esm__default = 'default';
const $e__index$$_$$cjs$$_$$in$$_$$esm__named = 'named';

const $n__index$$_$$cjs$$_$$in$$_$$esm = { exports: {default: $e__index$$_$$cjs$$_$$in$$_$$esm__default, named: $e__index$$_$$cjs$$_$$in$$_$$esm__named} };
})()
"
`;

exports[`flat packer with cjs in esm: target: cjs 1`] = `
"var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$cjs$$_$$exports */

let $n__dep$$_$$cjs$$_$$exports = { exports: {}};$n__dep$$_$$cjs$$_$$exports.exports.a = 1;
$n__dep$$_$$cjs$$_$$exports.exports.b = 2;

/* dep$$_$$cjs$$_$$module */

let $n__dep$$_$$cjs$$_$$module = { exports: {}};$n__dep$$_$$cjs$$_$$module.exports = {
  a: 1,
  b: 2
};

/* index$$_$$cjs$$_$$in$$_$$esm */





if ($n__dep$$_$$cjs$$_$$exports.exports.a < $n__dep$$_$$cjs$$_$$exports.exports.b) {
  console.log($n__dep$$_$$cjs$$_$$exports.exports.a, $n__dep$$_$$cjs$$_$$exports.exports.b);
}
if ($n__dep$$_$$cjs$$_$$module.exports.a < $n__dep$$_$$cjs$$_$$module.exports.b) {
  console.log($n__dep$$_$$cjs$$_$$module.exports);
}
if ($n__dep$$_$$cjs$$_$$exports.exports.a < $n__dep$$_$$cjs$$_$$exports.exports.b) {
  console.log($n__dep$$_$$cjs$$_$$exports.exports);
}

const $e__index$$_$$cjs$$_$$in$$_$$esm__default = 'default';
const $e__index$$_$$cjs$$_$$in$$_$$esm__named = 'named';

const $n__index$$_$$cjs$$_$$in$$_$$esm = { exports: {default: $e__index$$_$$cjs$$_$$in$$_$$esm__default, named: $e__index$$_$$cjs$$_$$in$$_$$esm__named} };
module.exports = $n__index$$_$$cjs$$_$$in$$_$$esm.exports;
"
`;

exports[`flat packer with cjs in esm: target: esm 1`] = `
"var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$cjs$$_$$exports */

let $n__dep$$_$$cjs$$_$$exports = { exports: {}};$n__dep$$_$$cjs$$_$$exports.exports.a = 1;
$n__dep$$_$$cjs$$_$$exports.exports.b = 2;

/* dep$$_$$cjs$$_$$module */

let $n__dep$$_$$cjs$$_$$module = { exports: {}};$n__dep$$_$$cjs$$_$$module.exports = {
  a: 1,
  b: 2
};

/* index$$_$$cjs$$_$$in$$_$$esm */





if ($n__dep$$_$$cjs$$_$$exports.exports.a < $n__dep$$_$$cjs$$_$$exports.exports.b) {
  console.log($n__dep$$_$$cjs$$_$$exports.exports.a, $n__dep$$_$$cjs$$_$$exports.exports.b);
}
if ($n__dep$$_$$cjs$$_$$module.exports.a < $n__dep$$_$$cjs$$_$$module.exports.b) {
  console.log($n__dep$$_$$cjs$$_$$module.exports);
}
if ($n__dep$$_$$cjs$$_$$exports.exports.a < $n__dep$$_$$cjs$$_$$exports.exports.b) {
  console.log($n__dep$$_$$cjs$$_$$exports.exports);
}

const $e__index$$_$$cjs$$_$$in$$_$$esm__default = 'default';
const $e__index$$_$$cjs$$_$$in$$_$$esm__named = 'named';

const $n__index$$_$$cjs$$_$$in$$_$$esm = { exports: {default: $e__index$$_$$cjs$$_$$in$$_$$esm__default, named: $e__index$$_$$cjs$$_$$in$$_$$esm__named} };
export default $e__index$$_$$cjs$$_$$in$$_$$esm__default;
export {$e__index$$_$$cjs$$_$$in$$_$$esm__named as named};
"
`;

exports[`flat packer with esm in cjs: target: app 1`] = `
"(function() {
var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$esm$$_$$named */

const $e__dep$$_$$esm$$_$$named__a = 1;
const $e__dep$$_$$esm$$_$$named__b = 2;

const $n__dep$$_$$esm$$_$$named = { exports: {a: $e__dep$$_$$esm$$_$$named__a, b: $e__dep$$_$$esm$$_$$named__b} };

/* dep$$_$$esm$$_$$default */

const $e__dep$$_$$esm$$_$$default__default = {
  a: 1,
  b: 2
};

const $n__dep$$_$$esm$$_$$default = { exports: {default: $e__dep$$_$$esm$$_$$default__default} };

/* index$$_$$esm$$_$$in$$_$$cjs */

let $n__index$$_$$esm$$_$$in$$_$$cjs = { exports: {}};const $i__index$$_$$esm$$_$$in$$_$$cjs__namedPair = ($n__dep$$_$$esm$$_$$named.exports);
const { a: $i__index$$_$$esm$$_$$in$$_$$cjs__a, b: $i__index$$_$$esm$$_$$in$$_$$cjs__b } = ($n__dep$$_$$esm$$_$$named.exports);
const $i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair = ($n__dep$$_$$esm$$_$$default.exports);

if ($i__index$$_$$esm$$_$$in$$_$$cjs__a < $i__index$$_$$esm$$_$$in$$_$$cjs__b) {
  console.log($i__index$$_$$esm$$_$$in$$_$$cjs__a, $i__index$$_$$esm$$_$$in$$_$$cjs__b);
}
if ($i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair.a < $i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair.b) {
  console.log($i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair);
}
if ($i__index$$_$$esm$$_$$in$$_$$cjs__namedPair.a < $i__index$$_$$esm$$_$$in$$_$$cjs__namedPair.b) {
  console.log($i__index$$_$$esm$$_$$in$$_$$cjs__namedPair);
}

$n__index$$_$$esm$$_$$in$$_$$cjs.exports.named = 'named';
})()
"
`;

exports[`flat packer with esm in cjs: target: cjs 1`] = `
"var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$esm$$_$$named */

const $e__dep$$_$$esm$$_$$named__a = 1;
const $e__dep$$_$$esm$$_$$named__b = 2;

const $n__dep$$_$$esm$$_$$named = { exports: {a: $e__dep$$_$$esm$$_$$named__a, b: $e__dep$$_$$esm$$_$$named__b} };

/* dep$$_$$esm$$_$$default */

const $e__dep$$_$$esm$$_$$default__default = {
  a: 1,
  b: 2
};

const $n__dep$$_$$esm$$_$$default = { exports: {default: $e__dep$$_$$esm$$_$$default__default} };

/* index$$_$$esm$$_$$in$$_$$cjs */

let $n__index$$_$$esm$$_$$in$$_$$cjs = { exports: {}};const $i__index$$_$$esm$$_$$in$$_$$cjs__namedPair = ($n__dep$$_$$esm$$_$$named.exports);
const { a: $i__index$$_$$esm$$_$$in$$_$$cjs__a, b: $i__index$$_$$esm$$_$$in$$_$$cjs__b } = ($n__dep$$_$$esm$$_$$named.exports);
const $i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair = ($n__dep$$_$$esm$$_$$default.exports);

if ($i__index$$_$$esm$$_$$in$$_$$cjs__a < $i__index$$_$$esm$$_$$in$$_$$cjs__b) {
  console.log($i__index$$_$$esm$$_$$in$$_$$cjs__a, $i__index$$_$$esm$$_$$in$$_$$cjs__b);
}
if ($i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair.a < $i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair.b) {
  console.log($i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair);
}
if ($i__index$$_$$esm$$_$$in$$_$$cjs__namedPair.a < $i__index$$_$$esm$$_$$in$$_$$cjs__namedPair.b) {
  console.log($i__index$$_$$esm$$_$$in$$_$$cjs__namedPair);
}

$n__index$$_$$esm$$_$$in$$_$$cjs.exports.named = 'named';
module.exports = $n__index$$_$$esm$$_$$in$$_$$cjs.exports;
"
`;

exports[`flat packer with esm in cjs: target: esm 1`] = `
"var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$esm$$_$$named */

const $e__dep$$_$$esm$$_$$named__a = 1;
const $e__dep$$_$$esm$$_$$named__b = 2;

const $n__dep$$_$$esm$$_$$named = { exports: {a: $e__dep$$_$$esm$$_$$named__a, b: $e__dep$$_$$esm$$_$$named__b} };

/* dep$$_$$esm$$_$$default */

const $e__dep$$_$$esm$$_$$default__default = {
  a: 1,
  b: 2
};

const $n__dep$$_$$esm$$_$$default = { exports: {default: $e__dep$$_$$esm$$_$$default__default} };

/* index$$_$$esm$$_$$in$$_$$cjs */

let $n__index$$_$$esm$$_$$in$$_$$cjs = { exports: {}};const $i__index$$_$$esm$$_$$in$$_$$cjs__namedPair = ($n__dep$$_$$esm$$_$$named.exports);
const { a: $i__index$$_$$esm$$_$$in$$_$$cjs__a, b: $i__index$$_$$esm$$_$$in$$_$$cjs__b } = ($n__dep$$_$$esm$$_$$named.exports);
const $i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair = ($n__dep$$_$$esm$$_$$default.exports);

if ($i__index$$_$$esm$$_$$in$$_$$cjs__a < $i__index$$_$$esm$$_$$in$$_$$cjs__b) {
  console.log($i__index$$_$$esm$$_$$in$$_$$cjs__a, $i__index$$_$$esm$$_$$in$$_$$cjs__b);
}
if ($i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair.a < $i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair.b) {
  console.log($i__index$$_$$esm$$_$$in$$_$$cjs__defaultPair);
}
if ($i__index$$_$$esm$$_$$in$$_$$cjs__namedPair.a < $i__index$$_$$esm$$_$$in$$_$$cjs__namedPair.b) {
  console.log($i__index$$_$$esm$$_$$in$$_$$cjs__namedPair);
}

$n__index$$_$$esm$$_$$in$$_$$cjs.exports.named = 'named';
"
`;

exports[`flat packer with only cjs: target: app 1`] = `
"(function() {
var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$cjs$$_$$exports */

let $n__dep$$_$$cjs$$_$$exports = { exports: {}};$n__dep$$_$$cjs$$_$$exports.exports.a = 1;
$n__dep$$_$$cjs$$_$$exports.exports.b = 2;

/* dep$$_$$cjs$$_$$module */

let $n__dep$$_$$cjs$$_$$module = { exports: {}};$n__dep$$_$$cjs$$_$$module.exports = {
  a: 1,
  b: 2
};

/* index$$_$$cjs */

let $n__index$$_$$cjs = { exports: {}};const $i__index$$_$$cjs__exportsPair = ($n__dep$$_$$cjs$$_$$exports.exports);
const { a: $i__index$$_$$cjs__a, b: $i__index$$_$$cjs__b } = ($n__dep$$_$$cjs$$_$$exports.exports);
const $i__index$$_$$cjs__modulePair = ($n__dep$$_$$cjs$$_$$module.exports);

if ($i__index$$_$$cjs__a < $i__index$$_$$cjs__b) {
  console.log($i__index$$_$$cjs__a, $i__index$$_$$cjs__b);
}
if ($i__index$$_$$cjs__modulePair.a < $i__index$$_$$cjs__modulePair.b) {
  console.log($i__index$$_$$cjs__modulePair);
}
if ($i__index$$_$$cjs__exportsPair.a < $i__index$$_$$cjs__exportsPair.b) {
  console.log($i__index$$_$$cjs__exportsPair);
}

$n__index$$_$$cjs.exports.named = 'named';
})()
"
`;

exports[`flat packer with only cjs: target: cjs 1`] = `
"var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$cjs$$_$$exports */

let $n__dep$$_$$cjs$$_$$exports = { exports: {}};$n__dep$$_$$cjs$$_$$exports.exports.a = 1;
$n__dep$$_$$cjs$$_$$exports.exports.b = 2;

/* dep$$_$$cjs$$_$$module */

let $n__dep$$_$$cjs$$_$$module = { exports: {}};$n__dep$$_$$cjs$$_$$module.exports = {
  a: 1,
  b: 2
};

/* index$$_$$cjs */

let $n__index$$_$$cjs = { exports: {}};const $i__index$$_$$cjs__exportsPair = ($n__dep$$_$$cjs$$_$$exports.exports);
const { a: $i__index$$_$$cjs__a, b: $i__index$$_$$cjs__b } = ($n__dep$$_$$cjs$$_$$exports.exports);
const $i__index$$_$$cjs__modulePair = ($n__dep$$_$$cjs$$_$$module.exports);

if ($i__index$$_$$cjs__a < $i__index$$_$$cjs__b) {
  console.log($i__index$$_$$cjs__a, $i__index$$_$$cjs__b);
}
if ($i__index$$_$$cjs__modulePair.a < $i__index$$_$$cjs__modulePair.b) {
  console.log($i__index$$_$$cjs__modulePair);
}
if ($i__index$$_$$cjs__exportsPair.a < $i__index$$_$$cjs__exportsPair.b) {
  console.log($i__index$$_$$cjs__exportsPair);
}

$n__index$$_$$cjs.exports.named = 'named';
module.exports = $n__index$$_$$cjs.exports;
"
`;

exports[`flat packer with only cjs: target: esm 1`] = `
"var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$cjs$$_$$exports */

let $n__dep$$_$$cjs$$_$$exports = { exports: {}};$n__dep$$_$$cjs$$_$$exports.exports.a = 1;
$n__dep$$_$$cjs$$_$$exports.exports.b = 2;

/* dep$$_$$cjs$$_$$module */

let $n__dep$$_$$cjs$$_$$module = { exports: {}};$n__dep$$_$$cjs$$_$$module.exports = {
  a: 1,
  b: 2
};

/* index$$_$$cjs */

let $n__index$$_$$cjs = { exports: {}};const $i__index$$_$$cjs__exportsPair = ($n__dep$$_$$cjs$$_$$exports.exports);
const { a: $i__index$$_$$cjs__a, b: $i__index$$_$$cjs__b } = ($n__dep$$_$$cjs$$_$$exports.exports);
const $i__index$$_$$cjs__modulePair = ($n__dep$$_$$cjs$$_$$module.exports);

if ($i__index$$_$$cjs__a < $i__index$$_$$cjs__b) {
  console.log($i__index$$_$$cjs__a, $i__index$$_$$cjs__b);
}
if ($i__index$$_$$cjs__modulePair.a < $i__index$$_$$cjs__modulePair.b) {
  console.log($i__index$$_$$cjs__modulePair);
}
if ($i__index$$_$$cjs__exportsPair.a < $i__index$$_$$cjs__exportsPair.b) {
  console.log($i__index$$_$$cjs__exportsPair);
}

$n__index$$_$$cjs.exports.named = 'named';
"
`;

exports[`flat packer with only esm: target: app 1`] = `
"(function() {
var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$esm$$_$$named */

const $e__dep$$_$$esm$$_$$named__a = 1;
const $e__dep$$_$$esm$$_$$named__b = 2;

const $n__dep$$_$$esm$$_$$named = { exports: {a: $e__dep$$_$$esm$$_$$named__a, b: $e__dep$$_$$esm$$_$$named__b} };

/* dep$$_$$esm$$_$$default */

const $e__dep$$_$$esm$$_$$default__default = {
  a: 1,
  b: 2
};

const $n__dep$$_$$esm$$_$$default = { exports: {default: $e__dep$$_$$esm$$_$$default__default} };

/* index$$_$$esm */





if ($e__dep$$_$$esm$$_$$named__a < $e__dep$$_$$esm$$_$$named__b) {
  console.log($e__dep$$_$$esm$$_$$named__a, $e__dep$$_$$esm$$_$$named__b);
}
if ($e__dep$$_$$esm$$_$$default__default.a < $e__dep$$_$$esm$$_$$default__default.b) {
  console.log($e__dep$$_$$esm$$_$$default__default);
}
if ($n__dep$$_$$esm$$_$$named.exports.a < $n__dep$$_$$esm$$_$$named.exports.b) {
  console.log($n__dep$$_$$esm$$_$$named.exports);
}

const $e__index$$_$$esm__default = 'default';
const $e__index$$_$$esm__named = 'named';

const $n__index$$_$$esm = { exports: {default: $e__index$$_$$esm__default, named: $e__index$$_$$esm__named} };
})()
"
`;

exports[`flat packer with only esm: target: cjs 1`] = `
"var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$esm$$_$$named */

const $e__dep$$_$$esm$$_$$named__a = 1;
const $e__dep$$_$$esm$$_$$named__b = 2;

const $n__dep$$_$$esm$$_$$named = { exports: {a: $e__dep$$_$$esm$$_$$named__a, b: $e__dep$$_$$esm$$_$$named__b} };

/* dep$$_$$esm$$_$$default */

const $e__dep$$_$$esm$$_$$default__default = {
  a: 1,
  b: 2
};

const $n__dep$$_$$esm$$_$$default = { exports: {default: $e__dep$$_$$esm$$_$$default__default} };

/* index$$_$$esm */





if ($e__dep$$_$$esm$$_$$named__a < $e__dep$$_$$esm$$_$$named__b) {
  console.log($e__dep$$_$$esm$$_$$named__a, $e__dep$$_$$esm$$_$$named__b);
}
if ($e__dep$$_$$esm$$_$$default__default.a < $e__dep$$_$$esm$$_$$default__default.b) {
  console.log($e__dep$$_$$esm$$_$$default__default);
}
if ($n__dep$$_$$esm$$_$$named.exports.a < $n__dep$$_$$esm$$_$$named.exports.b) {
  console.log($n__dep$$_$$esm$$_$$named.exports);
}

const $e__index$$_$$esm__default = 'default';
const $e__index$$_$$esm__named = 'named';

const $n__index$$_$$esm = { exports: {default: $e__index$$_$$esm__default, named: $e__index$$_$$esm__named} };
module.exports = $n__index$$_$$esm.exports;
"
`;

exports[`flat packer with only esm: target: esm 1`] = `
"var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* dep$$_$$esm$$_$$named */

const $e__dep$$_$$esm$$_$$named__a = 1;
const $e__dep$$_$$esm$$_$$named__b = 2;

const $n__dep$$_$$esm$$_$$named = { exports: {a: $e__dep$$_$$esm$$_$$named__a, b: $e__dep$$_$$esm$$_$$named__b} };

/* dep$$_$$esm$$_$$default */

const $e__dep$$_$$esm$$_$$default__default = {
  a: 1,
  b: 2
};

const $n__dep$$_$$esm$$_$$default = { exports: {default: $e__dep$$_$$esm$$_$$default__default} };

/* index$$_$$esm */





if ($e__dep$$_$$esm$$_$$named__a < $e__dep$$_$$esm$$_$$named__b) {
  console.log($e__dep$$_$$esm$$_$$named__a, $e__dep$$_$$esm$$_$$named__b);
}
if ($e__dep$$_$$esm$$_$$default__default.a < $e__dep$$_$$esm$$_$$default__default.b) {
  console.log($e__dep$$_$$esm$$_$$default__default);
}
if ($n__dep$$_$$esm$$_$$named.exports.a < $n__dep$$_$$esm$$_$$named.exports.b) {
  console.log($n__dep$$_$$esm$$_$$named.exports);
}

const $e__index$$_$$esm__default = 'default';
const $e__index$$_$$esm__named = 'named';

const $n__index$$_$$esm = { exports: {default: $e__index$$_$$esm__default, named: $e__index$$_$$esm__named} };
export default $e__index$$_$$esm__default;
export {$e__index$$_$$esm__named as named};
"
`;

exports[`flat packer with single js: target: app 1`] = `
"(function() {
var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* index$$_$$single */

const $i__index$$_$$single__a = 1;
if ($i__index$$_$$single__a < 2) {
  console.log($i__index$$_$$single__a);
}

const $e__index$$_$$single__default = 'default';
const $e__index$$_$$single__named = 'named';

const $n__index$$_$$single = { exports: {default: $e__index$$_$$single__default, named: $e__index$$_$$single__named} };
})()
"
`;

exports[`flat packer with single js: target: cjs 1`] = `
"var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* index$$_$$single */

const $i__index$$_$$single__a = 1;
if ($i__index$$_$$single__a < 2) {
  console.log($i__index$$_$$single__a);
}

const $e__index$$_$$single__default = 'default';
const $e__index$$_$$single__named = 'named';

const $n__index$$_$$single = { exports: {default: $e__index$$_$$single__default, named: $e__index$$_$$single__named} };
module.exports = $n__index$$_$$single.exports;
"
`;

exports[`flat packer with single js: target: esm 1`] = `
"var __DEV__ = false;
var __fastpack_cache__ = {};

function __fastpack_require__(f) {
  if (__fastpack_cache__[f.name] === undefined) {
    __fastpack_cache__[f.name] = f();
  }
  return __fastpack_cache__[f.name];
}

function __fastpack_import__(f) {
  return new Promise((resolve, reject) => {
    try {
      resolve(__fastpack_require__(f));
    } catch (e) {
      reject(e);
    }
  });
}

/* index$$_$$single */

const $i__index$$_$$single__a = 1;
if ($i__index$$_$$single__a < 2) {
  console.log($i__index$$_$$single__a);
}

const $e__index$$_$$single__default = 'default';
const $e__index$$_$$single__named = 'named';

const $n__index$$_$$single = { exports: {default: $e__index$$_$$single__default, named: $e__index$$_$$single__named} };
export default $e__index$$_$$single__default;
export {$e__index$$_$$single__named as named};
"
`;
